src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/SimpleConfigModel.cs

```diff
@@ -559,56 +575,54 @@ private IGlobalRunConfig ProduceOperationSetup(SpecialOperationConfigParams defa
+                    else
+                    {
+                        //Replace the literal value in the "replaces" property with the evaluated value of the symbol
+                        sourceVariable = symbol.Key;
+                    }
 
-                        if (sourceVariable != null)
+                    if (sourceVariable != null)
+                    {
+                        if (symbol.Value is BaseValueSymbol p)
                         {
-                            if (symbol.Value is BaseValueSymbol p)
+                            foreach (string formName in p.Forms.GlobalForms)
                             {
-                                foreach (string formName in p.Forms.GlobalForms)
+                                if (Forms.TryGetValue(formName, out IValueForm valueForm))
                                 {
-                                    if (Forms.TryGetValue(formName, out IValueForm valueForm))
+                                    string symbolName = sourceVariable + "{-VALUE-FORMS-}" + formName;
+                                    if (!string.IsNullOrWhiteSpace(symbol.Value.Replaces))
                                     {
-                                        string symbolName = symbol.Key + "{-VALUE-FORMS-}" + formName;
                                         string processedReplacement = valueForm.Process(Forms, p.Replaces);
                                         GenerateReplacementsForParameter(symbol, processedReplacement, symbolName, macroGeneratedReplacements);
-
-                                        if (generateMacros)
-                                        {
-                                            macros.Add(new ProcessValueFormMacroConfig(symbol.Key, symbolName, "string", formName, Forms));
-                                        }
                                     }
-                                    else
+                                    if (generateMacros)
                                     {
-                                        EnvironmentSettings.Host.LogDiagnosticMessage($"Unable to find a form called '{formName}'", "Authoring");
+                                        macros.Add(new ProcessValueFormMacroConfig(sourceVariable, symbolName, "string", formName, Forms));
                                     }
                                 }
+                                else
+                                {
+                                    EnvironmentSettings.Host.LogDiagnosticMessage($"Unable to find a form called '{formName}'", "Authoring");
+                                }
                             }
-                            else
-                            {
-                                GenerateReplacementsForParameter(symbol, symbol.Value.Replaces, sourceVariable, macroGeneratedReplacements);
-                            }
+                        }
+                        else if (!string.IsNullOrWhiteSpace(symbol.Value.Replaces))
```

Comment:
minor, previously this call happened as a default to the if statements.  Now the call happens in an explicit case & nothing is done in the "else" case.  Can you confirm there is no required logic to be run in the else?

Reply:
previously check for `symbol.Value.Replaces` was done outside of if-else block, however some parts of that logic is needed not only for calculating replacements but also for generating the macro operations. Now check for symbol.Value.Replaces is done only to the blocks which needed to calculate replacements only.